scan/src/subscriptions/TxSub.re

```
@@ -282,7 +296,17 @@ module Msg = {
       }
     );
   };
-  let decodeActions = json => json |> JsonUtils.Decode.list(decodeAction);
+
+  let decodeFailAction = json => FailMessage(json |> FailMessage.decode);
+
+  let decodeActions = json =>
+    JsonUtils.Decode.(
+      switch (json |> field("status", string)) {
+      | "success" => json |> field("messages", list(decodeAction))
+      | "fail" => json |> field("messages", list(decodeFailAction))
+      | _ => []
```

Let's do `exception` so we fail fast.